src/replica/replica_2pc.cpp

```diff
@@ -490,21 +490,15 @@ void replica::on_prepare(dsn::message_ex *request)
     error_code err = _prepare_list->prepare(mu, status());
     dassert(err == ERR_OK, "prepare mutation failed, err = %s", err.to_string());
 
-    if (partition_status::PS_POTENTIAL_SECONDARY == status()) {
+    if (partition_status::PS_POTENTIAL_SECONDARY == status() ||
+        partition_status::PS_SECONDARY == status()) {
         dassert(mu->data.header.decree <=
                     last_committed_decree() + _options->max_mutation_count_in_prepare_list,
                 "%" PRId64 " VS %" PRId64 "(%" PRId64 " + %d)",
                 mu->data.header.decree,
                 last_committed_decree() + _options->max_mutation_count_in_prepare_list,
                 last_committed_decree(),
                 _options->max_mutation_count_in_prepare_list);
-    } else if (partition_status::PS_SECONDARY == status()) {
```

Comment:
I remove this dassert because the secondary commit async, and this will make this dassert not satified.